src/status_im/contexts/wallet/events.cljs

```diff
@@ -355,19 +357,61 @@
 
 (rf/reg-event-fx :wallet/bridge-select-token
- (fn [{:keys [db]} [{:keys [token token-symbol stack-id start-flow?]}]]
-   (let [missing-recipient? (-> db :wallet :ui :send :to-address nil?)
-         to-address         (-> db :wallet :current-viewing-account-address)]
+ (fn [{:keys [db]}
+      [{:keys [token token-symbol stack-id network owners network-details start-flow?] :as params}]]
+   (let [{:keys [wallet]}       db
+         unique-owner           (when (= (count owners) 1)
+                                  (first owners))
+         token                  (if (and unique-owner (nil? token))
+                                  (let [token (utils/get-token-from-account db
+                                                                            token-symbol
+                                                                            unique-owner)]
+                                    (assoc token
+                                           :networks (network-utils/network-list-with-positive-balance
+                                                      token
+                                                      network-details)
+                                           :supported-networks (network-utils/network-list
+                                                                token
+                                                                network-details)
+                                           :total-balance (utils/calculate-total-token-balance token)))
+                                  token)
+         missing-recipient?     (-> db :wallet :ui :send :to-address nil?)
+         to-address             (or unique-owner
+                                    (-> db :wallet :current-viewing-account-address)
+                                    (utils/get-default-account (-> db :wallet :accounts)))
+         view-id                (:view-id db)
+         root-screen?           (or (= view-id :wallet-stack) (nil? view-id))
+         multi-account-balance? (-> (utils/get-accounts-with-token-balance (:accounts wallet)
+                                                                           token)
+                                    (count)
+                                    (> 1))]
      {:db (cond-> db
             :always            (assoc-in [:wallet :ui :send :tx-type] :tx/bridge)
             token              (assoc-in [:wallet :ui :send :token] token)
             token-symbol       (assoc-in [:wallet :ui :send :token-symbol] token-symbol)
+            network            (assoc-in [:wallet :ui :send :network] network)
             missing-recipient? (assoc-in [:wallet :ui :send :to-address] to-address))
-      :fx [[:dispatch
-            [:wallet/wizard-navigate-forward
-             {:current-screen stack-id
-              :start-flow?    start-flow?
-              :flow-id        :wallet-bridge-flow}]]]})))
+      :fx (if (and (not unique-owner) multi-account-balance? root-screen?)
```

Comment:
The logic of this `:fx` is very hard to comprehend because of nested `if`s. I'd suggest to rewrite it to `cond` and comment every clause.

Reply:
Will do!

Reply:
Done